cmake/Modules/CMakePolicy.cmake

```diff
@@ -37,4 +37,13 @@ endif()
 # building with the Apple linker.
 if(POLICY CMP0156)
   cmake_policy(SET CMP0156 NEW)
+
+  # CMP0179: De-duplication of static libraries on link lines keeps first occurrence.
+  # Dependent on CMP0156=NEW. Keeps the first occurrence of static libraries
+  # during deduplication to unify the behaviour on all platforms.
+  # Also works around LLD bug exhibited since CMake 3.31.0 with an expected
+  # workaround in 3.31.1 (https://gitlab.kitware.com/cmake/cmake/-/issues/26447).
+  if(POLICY CMP0179)
+    cmake_policy(SET CMP0179 NEW)
+  endif()
```

Comment:
I think that this isn't the ideal way to handle this. I think that we should simply limit CMP0156 to `if(APPLE OR WIN32)`. ELFish platforms have the expectation that the command line order and duplication is material and the de-duplication will change the behaviour of the symbolic resolution.

Reply:
These CMake targets will eventually be set to new (unless explicitly set to old) once we bump the minimum required CMake version. It would be nice if when we run across issues we set the policy to make it easier when bumping the CMake version.

Agreed that this is not really the correct solution to the specific problem mentioned here though.

Reply:
I don't see a reason to limit this platform-wise. It works fine on all up until CMake 3.31.0 started taking advantage of leeway granted by CMP0156=NEW specification, if I understand correctly. However LLVM is affected by this and requires the more sensible behaviour provided by CMP0179=NEW, which coincides with the above change.

> Agreed that this is not really the correct solution to the specific problem mentioned here though.

What would be the correct solution?  
CMP0179 solves the problem, it'll eventually get set anyway across the board and it changes to behaviour to what LLVM requires to be properly linked.